aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -73,10 +73,19 @@ class CAFFE2_API Dispatcher final {
   c10::optional<OperatorHandle> findSchema(const OperatorName& operator_name);
 
   /**
-   * Variant of findSchema that results in less code generated at the call site
-   * if you have string literals.
+   * Variant of findSchema that results in less code generated at the call site.
+   * It (1) takes const char* pointer rather than OperatorName (so we skip
+   * generating std::string constructor calls at the call site), and (2)
+   * it raises an exception if the operator is not found (so we skip
+   * generating exception raising code at the call site)
+   *
+   * Irritatingly, we still have to generate the handful of instructions
```

Comment:
did you test returning `c10::optional` and then making it `noexcept`?

Reply:
No, but I know what it looks like: you skip the guard failure handling (a few instructions) and bring back exception raising (a lot of instructions).

Reply:
(or maybe you don't even get to skip guard failure, that seems likely to me)